public/app/features/transformers/extractFields/fieldExtractors.ts

```diff
@@ -1,20 +1,43 @@
-import { Registry, RegistryItem } from '@grafana/data';
+import { Registry, RegistryItem, stringStartsAsRegEx, stringToJsRegex } from '@grafana/data';
 
-import { FieldExtractorID } from './types';
+import { ExtractFieldsOptions, FieldExtractorID } from './types';
+
+type Parser = (v: string) => Record<string, any> | undefined;
 
 export interface FieldExtractor extends RegistryItem {
-  parse: (v: string) => Record<string, any> | undefined;
+  getParser: (opts: ExtractFieldsOptions) => Parser;
 }
 
 const extJSON: FieldExtractor = {
   id: FieldExtractorID.JSON,
   name: 'JSON',
   description: 'Parse JSON string',
-  parse: (v: string) => {
+  getParser: (options) => (v: string) => {
     return JSON.parse(v);
   },
 };
 
+const extRegExp: FieldExtractor = {
+  id: FieldExtractorID.RegExp,
+  name: 'RegExp',
+  description: 'Parse with RegExp',
+  getParser: (options) => {
+    let regex: RegExp | null = /(?<NewField>.*)/;
+
+    if (stringStartsAsRegEx(options.regExp!)) {
```

Comment:
You're loving this `stringStartsAsRegEx` function lately! 😆

Reply:
just trying to recycle what we already have. unfortunately i have to keep using this pre-check since `stringToJsRegex()` will convert whatever you pass it to a regexp without escaping :(